### DISCUSSION TITLE
Host Permissions based on Manifest version

### DISCUSSION THREAD

**Discussion by nnx0r** [wxt-dev/wxt#219] [Q&A] [answered]
I thought that building the extension will automatically read manifest configuration and check if there's `host_permissions` property and merge it with `permissions` for v2. 

<img width="634" alt="Screenshot 2023-11-07 at 23 01 19" src="https://github.com/wxt-dev/wxt/assets/3460368/8b3d0e3c-d4dc-499f-8a91-ca50757fa11a">

Now that I think about it, there are many differences between v2 and v3, so magically doing this for permissions might be a bad idea.

Can you tell me if this is correct approach?

```ts
export default defineConfig({
  manifest: ({ manifestVersion }) => {
    const hosts = ['https://*/*', 'http://*/*']

    const config: UserManifest = {
      permissions: ['activeTab', 'storage', 'webNavigation'],
      host_permissions: hosts,
    }

    if (manifestVersion === 2) {
      config.permissions!.push(...hosts)
      delete config.host_permissions
    }

    return config
  },
})
```

**Comment by aklinker1** [comment, marked as answer]
Yes, this is correct.
